src/shared_realm.hpp

```diff
@@ -247,10 +247,10 @@ class Realm : public std::enable_shared_from_this<Realm> {
         };
 
         VersionID m_version_id;
-        std::vector<_impl::AnyHandover> m_objects;
+        std::unique_ptr<_impl::AnyHandover> m_handover; // FIXME: Remove this extra layer of indirection.
```

Comment:
There are a few things that I don't _love_ about this implementation, but I'm not sure how to improve. I don't like that this extra `unique_ptr` indirection is necessary, but removing it requires we drag in HUGE headers. I also don't love that we've kept `AnyHandover` and `AnyThreadConfined`, robbing the Object Store API of type safety. It would be better to use a template here, BUT without the extra indirection provided by `AnyHandover`, we'd have to drag in HUGE headers. Also, this would simply place more burden on the binding; the Objective-C API would have to use an abstract class with concrete subclasses for each handover type. I think this is a better design, but it's problematic in the ways previously discussed.